### DISCUSSION TITLE
Is there SIMD support already implemented or planned for software rendering?

### DISCUSSION THREAD

**Discussion by voloder** [slint-ui/slint#5627] [General]
Hello!

The reason why I'm asking this is because I'm interested in improving Slint's software rendering performance, more specifically on an ARM Neon based device. Does the Rust compiler automatically vectorize all instructions? Would it make sense to use the nightly `std::simd` crate in draw functions?

**Comment by tronical** [comment]
I think SIMD support for the Slint software renderer would be great to have. cc @ogoffart . `std::simd` looks very tempting, it's a bummer that it's nightly only :-(. So if we take `std::simd` into us, it would be great if that could be behind some kind of gate so that the code continues to compile without.

**Comment by ogoffart** [comment]
yes, SIMD support in the software renderer would be welcome.
I think to do that, you would need to extend the TargetPixel trait with functions like `blend_image_...` or something. At least that's more or less what i had in mind.
